SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -739,14 +759,34 @@ int64 USpatialActorChannel::ReplicateActor()
 	if (NetDriver->LoadBalanceStrategy != nullptr &&
 		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
+		if (MigrationVirtualWorkerId != SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+		{
+			if (NetDriver->LockingPolicy->IsLocked(Actor))
+			{
+				AActor* HierarchyRoot = SpatialGDK::GetHierarchyRoot(Actor);
+				DropActorHierarchyMigration(NetDriver, HierarchyRoot);
```

Comment:
Do we need to migrate them all at once? Does this break if the user changes the locking policy?

Reply:
I expect it to break if another locking policy does not respect ownership.